app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java

```diff
@@ -262,26 +265,26 @@ private void createAddressView() {
      * Registering user to the activity document Firebase Firestore (field array "participantId")
      */
     public void registerToActivity(View v) {
+        registerToActivityImplementation(activity, userId);
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    public void registerToActivityImplementation(Activity activity, String userId) {
         if (!activity.getParticipantId().contains(userId)) {
             try {
                 activity.addParticipantId(userId);
-                createParticipantNumberView();
-                activityManager.update(activity.getDocumentPath(), "participantId", userId, "union").addOnSuccessListener(new OnSuccessListener() {
+                createParticipantNumberView(activity);
+                // Adding the activity path to the array field "registeredActivity" of the Firebase
+                // Firestore user document
+                userManager.update(FirestoreUserManager.USERS_COLLECTION + ImageHandler.PATH_SEPARATOR + userId, REGISTERED_ACTIVITY_FIELD, activity.getDocumentPath(), UPDATE_FIELD_UNION);
+                // Adding the user ID to the array field "participantId" of the Firebase Firestore
+                // activity document
+                activityManager.update(activity.getDocumentPath(), PARTICIPANT_ID_FIELD, userId, UPDATE_FIELD_UNION).addOnSuccessListener(new OnSuccessListener() {
                     @Override
                     public void onSuccess(Object o) {
                         Log.d(TAG, "Participant registered in Firebase Firestore!");
-                        getParticipantNames();
-                        setButton();
-                        // TODO: (By Victor) here --> get activity from the MainMapFragment and update it!
```

Comment:
The todo is done ?

Reply:
Yes, it is done! And in fact, he just released an optimised version of the "reloading of the activities" based on a "Listener" no later than today!